File: apps/fc-app/code/pages/application_page.py
from shiny import ui

_CSS = """
body { display: flex; flex-direction: column; min-height: 100vh; }
.bslib-page-navbar > .tab-content { flex: 1; min-height: 0; padding: 0; margin: 0; }
textarea#data_path { font-family: monospace; font-size: 12px; color: #31387d; }
.btn { padding: 6px 10px; }
"""

_JS = """
function sendButtonID(id) {
    Shiny.setInputValue('last_cell_clicked', id, {priority: 'event'});
}
"""


def application_page():
    return ui.nav_panel(
        "Flow Cytometry Analysis",
        ui.tags.style(_CSS),
        ui.tags.script(_JS),
        ui.layout_sidebar(
            ui.sidebar(
                ui.p(ui.tags.small("Flow Cytometry Analysis App"), class_="text-muted"),
                ui.input_text_area(
                    "data_path",
                    "Project Folder Path",
                    placeholder="/path/to/project",
                    rows=4,
                ),
                ui.input_action_button(
                    "read_data",
                    "Read Flow Cytometry Data",
                    class_="btn-secondary w-100",
                ),
                width=290,
                open="always",
                style="background-color: rgba(238,238,238,1);",
            ),
            ui.div(
                ui.div(id="main_contents"),
                ui.div(id="main_contents1"),
                style="padding: 12px;",
            ),
        ),
    )

File: apps/tlgs-app/code/ui_pages/application_page.R
# author: Kwame Okrah
# date: 2026-03-05

main_contents = tags$div(
    tags$div(id = "main_contents"),
)

app_card = tags$div(
    style="height: 860px; font-size: 14px;",

    layout_sidebar(
      height = "100%",
      border_color = "rgba(227, 227, 227, 1)",
      bg = "rgba(255, 255, 255, 1)",

      sidebar = sidebar(
                    id = "my_sidebar",
                    width = 250,
                    open = "open",
                    bg = "rgba(238, 238, 238, 1)",

                    tags$p(paste0("Today's date: ", Sys.Date()),
                           class="text-secondary"),
                    
                    tags$div(id = "card_div_top")),
      
      main_contents
    )
)

application_page = bslib::nav_panel(
    tags$style(
        HTML("
            textarea.form-control {
                font-size: 13px;
            }

            .btn {
                padding: 6px 10px;
                font-size: 14px;
            }

            .cell-btn:hover {
                color: white;
                font-weight: bold;
                cursor: pointer;
            }

            #data_path {
                color: #017BC2;
            }

            /* Sidebar select dropdown */
            .bslib-sidebar-layout > .sidebar select.form-select {
                font-size: 13px;
            }

            /* Per-column filter inputs (selectize) above the projects
               table. Active-filter state tints the whole input blue so
               it stands out from inactive filters at a glance. */
            #projects_filter_ui .selectize-input.focus {
                border-color: #86b7fe !important;
                box-shadow: 0 0 0 0.2rem rgba(13,110,253,.15) !important;
            }
            #projects_filter_ui .selectize-input.has-items {
                border-color: #0d6efd !important;
                background-color: #e7f1ff !important;
                box-shadow: 0 0 0 0.15rem rgba(13,110,253,.15) !important;
            }
        ")
    ),

    title = "Flow Cytometry TLGs",
    page_banner("Make flow cytometry tables, graphs, and listings"),

    app_card
)
